### DISCUSSION TITLE
Authentication error after Keycloak upgrade [resolved]

### DISCUSSION THREAD

**Discussion by vc43** [outline/outline#4566] [Self Hosting]
There was an error after the upgrade Keycloak from version 11 to 20:
```
{"error":"invalid json response body at https://.../auth/realms/.../protocol/openid-connect/userinfo reason: Unexpected end of JSON input","stack":"FetchError: invalid json response body at https://.../auth/realms/.../protocol/openid-connect/userinfo reason: Unexpected end of JSON input\n    at /opt/outline/node_modules/node-fetch/lib/index.js:273:32\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async OAuth2Strategy._passportOauth.Strategy.userProfile (/opt/outline/build/server/routes/auth/providers/oidc.js:49:22)","level":"error","message":"Error during authentication"}
```
All other Keycloak clients work fine.
Outline version 0.62.

**Comment by vc43** [comment]
I found an answer [here](https://keycloak.discourse.group/t/issue-on-userinfo-endpoint-at-keycloak-20/18461). There are some changes in Keycloak 20 so it's necessary to create in Keycloak a new Client Scope named "openid" and add it on the Client page in tab Client Scopes as default.

**Comment by dannykorpan** [comment]
Same issue here. The solution was
OIDC_SCOPES=openid profile email

**Reply by jeffshead** [reply]
> The solution was OIDC_SCOPES=openid profile email

This works! With this solution, you can simply edit your Portainer stack/config file. The alternative is to add `openid` to the KeyCloack Client Scopes and set the new `openid` client scope as a default in the Outline client, in KeyCloak.
